### DISCUSSION TITLE
Using Perses with TLS and Prometheus with plain HTTP

### DISCUSSION THREAD

**Discussion by indigo423** [perses/perses#2171] [Q&A]
# Problem
When you run Perses using TLS and your time series backend with Prometheus is just plain HTTP you can configure the data source with the "HTTP Settings: Direct access". Saving the data source is indicated successfully. When you try now to create some dashboards you will see an "Load failed" error message without further indication of what the real problem is.
 
<img width="1418" alt="Screenshot 2024-08-14 at 00 21 38" src="https://github.com/user-attachments/assets/02dade72-5696-4b47-87a3-7e500d3ef74a">

# Diagnostics
When you use the browser inspector and look at the JavaScript console output you will find the error message: "Fetch API cannot load http://<your-server>:9009/prometheus/api/v1/guery_range due to access control checks."

<img width="884" alt="Screenshot 2024-08-14 at 00 17 08" src="https://github.com/user-attachments/assets/8752a9d4-08dc-4ec6-8cee-5654a208bcdb">

# Solution

If you have a requirement to run Perses and Prometheus in a mixed HTTPS/HTTP mode you need to configure Perses as a Proxy for the fetch queries.

<img width="1057" alt="Screenshot 2024-08-14 at 00 28 50" src="https://github.com/user-attachments/assets/e772aede-91d0-4b47-945e-e419ae8c0055">

**Comment by Nexucis** [comment]
closing this as it has been written as a troubleshooting documentation
